### ISSUE TITLE
Multiplication of arrays fails unexpectedly

### ISSUE THREAD

**Issue by rennis250** [arrayfire/arrayfire-rust#122]
The following code works:

```rust
extern crate arrayfire as af;

use af::*;

fn main() {
    set_backend(Backend::CPU);

    let a = Array::new(&vec![20f32; 100], Dim4::new(&[10, 10, 1, 1]));
    let b = Array::new(&vec![10f32; 100], Dim4::new(&[10, 10, 1, 1]));

    let c = a * b;
    print(&c);
}
```

The following does not:

```rust
extern crate arrayfire as af;

use af::*;

fn main() {
    set_backend(Backend::CPU);

    let a = Array::new(&vec![20f32; 110], Dim4::new(&[11, 10, 1, 1]));
    let b = Array::new(&vec![10f32; 100], Dim4::new(&[10, 10, 1, 1]));

    let c = a * b; // fails with incorrect size error
    print(&c);
}
```

I don't understand why.

Best,
Rob

**Comment by pavanky** [comment]
How can you possibly multiple two arrays of different dimensions ? What do you expect the output to be ?

**Comment by rennis250** [comment]
Shouldn't array multiplication follow the usual rules: If array1 has size MxN and array2 has size NxM, then array3 has size MxM. So, I would expect a resulting array of size 11x10 from the second example, but since I cannot create arrays of size = (10, 10, 0, 0) or size = (11, 10, 0, 0) from slices, I'm having trouble determining how I construct arrays in ArrayFire that I can succesfully multiply together.

**Comment by rennis250** [comment]
Oh, I see. The ```*``` operator is not shorthand for the ```matmul()``` function, which I just found. Sorry for the noise, but then, what does ```*``` for an array mean in the Rust library? What elements does it work with? I'll take a closer look at the documentation and code.

**Comment by pavanky** [comment]
@rennis250 It sounds like you are coming from a matlab background. The `*` performs the equivalent of `.*` from matlab. The same is true for all mathematical operations. (i.e. `/` is equivalent to `./` etc).

**Comment by rennis250** [comment]
Yes, I see now. I'll spend more time reading the original ArrayFire documentation. Sorry about that!

Best,
Rib

**Comment by pavanky** [comment]
@9prady9 I can't seem to find the documentation for the operators. Can you point it out ?

**Comment by 9prady9** [comment]
Operators in rust are trait implements for Array Object. So we can find the traits implemented on Array struct in Array page. You can find them over the below URL.

http://arrayfire.org/arrayfire-rust/arrayfire/struct.Array.html

**Comment by rennis250** [comment]
@9prady9: I saw that, but the documentation there didn't make it clear to me that ```*``` does elementwise multplication. Perhaps a short note at the top of the README which states that the Rust operators mimic the C++ operators, as well as a link to the C++ documentation? To be honest, it was my own fault to not consider the C++ documentation while reading the Rust documentation, but I suppose there will be others that fall into similar traps.

**Comment by 9prady9** [comment]
@rennis250 Sure, will take care of it.
